tests/platform_tests/mellanox/util.py

```diff
@@ -0,0 +1,329 @@
+            'RX1Power': '0.927dBm',
+            'RX2Power': '0.938dBm',
+            'RX3Power': '0.912dBm',
+            'RX4Power': '0.95dBm',
+            'TX1Bias': '6.75mA',
+            'TX1Power': '1.071dBm',
+            'TX2Bias': '6.75mA',
+            'TX2Power': '1.04dBm',
+            'TX3Bias': '6.75mA',
+            'TX3Power': '1.039dBm',
+            'TX4Bias': '6.75mA',
+            'TX4Power': '1.031dBm'
+        },
+        'ChannelThresholdValues': {
+            'RxPowerHighAlarm': '5.4dBm',
+            'RxPowerHighWarning': '2.4dBm',
+            'RxPowerLowAlarm': '-13.307dBm',
+            'RxPowerLowWarning': '-10.301dBm',
+            'TxBiasHighAlarm': '8.5mA',
+            'TxBiasHighWarning': '8.0mA',
+            'TxBiasLowAlarm': '5.492mA',
+            'TxBiasLowWarning': '6.0mA'
+        },
+        'ModuleMonitorValues': {
+            'Temperature': '43.105C',
+            'Vcc': '3.235Volts'
+        },
+        'ModuleThresholdValues': {
+            'TempHighAlarm': '80.0C',
+            'TempHighWarning': '70.0C',
+            'TempLowAlarm': '-10.0C',
+            'TempLowWarning': '0.0C',
+            'VccHighAlarm': '3.5Volts',
+            'VccHighWarning': '3.465Volts',
+            'VccLowAlarm': '3.1Volts',
+            'VccLowWarning': '3.135Volts'
+        }
+    }
+    """
+    one_sfp_eeprom_info_dict = {}
+    second_layer_dict = {}
+    previous_key = ""
+    top_key = ""
+    for line in sfp_eeprom_info.split("\n"):
+        res1 = re.match(pattern_top_layer_key_value, line)
+        if res1:
+            top_key = res1.groupdict()["key"].strip()
+            one_sfp_eeprom_info_dict[top_key] = res1.groupdict()["value"].strip()
+            continue
+        res2 = re.match(pattern_second_layer_key_value, line)
+        if res2:
+            if second_layer_dict and previous_key:
+                one_sfp_eeprom_info_dict[previous_key] = second_layer_dict
+                second_layer_dict = {}
+            one_sfp_eeprom_info_dict[res2.groupdict()["key"].strip()] = res2.groupdict()["value"].strip()
+            previous_key = res2.groupdict()["key"].strip()
+        else:
+            res3 = re.match(pattern_third_layer_key_value, line)
+            if res3:
+                second_layer_dict[res3.groupdict()["key"].strip()] = res3.groupdict()["value"].strip()
+    if second_layer_dict and previous_key:
+        one_sfp_eeprom_info_dict[previous_key] = second_layer_dict
+
+    return top_key, one_sfp_eeprom_info_dict
+
+
+def parse_sfp_eeprom_infos(eeprom_infos):
+    """
+    This method is to pares sfp eeprom infos, and return sfp_eeprom_info_dict
+    """
+    sfp_eeprom_info_dict = {}
+    for sfp_info in eeprom_infos.split("\n\n"):
+        intf, eeprom_info = parse_one_sfp_eeprom_info(sfp_info)
+        sfp_eeprom_info_dict[intf] = eeprom_info
+    return sfp_eeprom_info_dict
+
+
+def check_sfp_eeprom_info(duthost, sfp_eeprom_info, is_support_dom, show_eeprom_cmd):
```

Comment:
Could you please consider using compare the output of show eeprom command with ethtool/mlxlink output?

Reply:
for a specific field, the output from ethtool/mlxlink could have some differences with the SONiC CLI output, no easy way to directly compare of them.  only the Vendor OUI, Vendor Name, Vendor SN.. are the same and can be compared directly.

Reply:
Will use mlxlink tool to check if it supports dom.